specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/CommonDefinitions.json

```diff
@@ -3570,6 +3574,54 @@
         }
       }
     },
+    "DaprConfig": {
+      "description": "App Dapr configuration.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "Boolean indicating if the Dapr side car is enabled",
+          "type": "boolean",
```

Comment:
>boolean [](http://example.com/codeflow?start=19&length=7)

ARM recommends enums over booleans for future proof APIs [for new properties].

Standard guidance is: replace boolean/switch properties with better enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum is more flexible and future proof option because it allows additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].. Also, properties should always ideally provide more meaningful values just than True and False, which are meaningless without context. For example two boolean switches "isTypeA" and "isTypeB" can be replaced with one enum "type": [A, B, DefaultType].

Note: do NOT just define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its still confusing, thanks to the boolean value names."

Reply:
Thanks @mnasnh1009 , these Dapr properties are passed from Microsoft Web to Microsoft App RP and App RP today has "enabled" as boolean value - https://github.com/Azure/azure-rest-api-specs/blob/0ab5469dc0d75594f5747493dcfe8774e22d728f/specification/app/resource-manager/Microsoft.App/stable/2022-10-01/ContainerApps.json#L649C1-L649C1. Making "enabled" an enum in web RP will not be useful as the value from here is always passed to the APP RP and can't exist or be used alone.